ansible_base/api_documentation/postprocessing_hooks.py

```diff
@@ -0,0 +1,435 @@
+        return False
+
+    prefix = operation_id.split('_')[0] if '_' in operation_id else operation_id
+    if prefix in SKIP_AI_DESCRIPTION_PREFIXES:
+        return True
+    return False
+
+
+def _lookup_resource_purpose(operation_id):
+    """Look up resource_purpose for an operation via its prefix and ViewSet class."""
+    prefix = extract_operation_prefix(operation_id)
+    class_info = OPERATION_CLASS_MAP.get(prefix)
+
+    if not class_info:
+        return None, None
+
+    class_name, _, _ = class_info  # Extract class name from (class_name, path_parts_count, path_parts) tuple
+    resource_purpose = RESOURCE_PURPOSE_MAP.get(class_name)
+    return resource_purpose, class_name
+
+
+def _generate_from_resource_purpose(resource_purpose, action):
+    """Generate description from resource_purpose if action is supported."""
+    if not resource_purpose:
+        return None
+
+    if action in OPERATION_DESCRIPTION_TEMPLATES.keys():
+        return generate_description_from_purpose(resource_purpose, action)
+
+    return None
+
+
+def _generate_description_auto(operation_id, action, resource_parts, parent_resource, path, operation):
+    """Auto-generate description from resource names and operation type."""
+    # Format the resource name with proper grammar
+    resource_name = format_compound_resource(resource_parts, parent_resource, action)
+
+    # Get the descriptive operation name
+    operation_name = OPERATION_DESCRIPTIVE_NAMES.get(action, action.replace('_', ' ').capitalize())
+
+    # Handle associate/disassociate operations
+    if operation_id.endswith('_associate_create') or operation_id.endswith('_disassociate_create'):
+        return generate_associate_description(operation_id, path, resource_name)
+
+    # Try standard CRUD operation
+    ai_description = generate_crud_description(action, operation_name, resource_name, parent_resource)
+
+    # Fall back to custom action if not CRUD
+    if ai_description is None:
+        ai_description = generate_custom_action_description(operation_name, resource_name, operation)
+
+    return ai_description
+
+
+def _enforce_character_limit(description, max_length=300):
+    """Enforce character limit on description, truncating if necessary."""
+    if len(description) > max_length:
+        return description[: max_length - 3] + "..."
+    return description
+
+
+def _process_operation(operation, method, path):
+    """Process a single operation to add x-ai-description (modifies operation in-place)."""
+    # Skip if already has x-ai-description (respect explicit definitions)
+    if 'x-ai-description' in operation:
+        logger.debug(f"x-ai-description already defined for {path} {method.upper()}. Respecting existing value.")
+        return
+
+    # Get operation ID
+    operation_id = operation.get('operationId', '')
+
+    # Check if the ViewSet has opted out of AI description generation
+    if _should_skip_ai_description(operation_id):
+        return
```

Comment:
Add a debug here as to why you are just returning to save yourself some trouble shooting headache in the future.